### PR TITLE
chore: log pod identity response for error

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

Logs the response which contains error with pod identity.

Example:
```bash
E0623 17:59:23.823541       1 server.go:55] "failed to process mount request" err="failed to get keyvault client: failed to get key vault token: nmi response failed with status code: 404, err: getting assigned identities for pod default/nginx-b64f4d687-jnppl in CREATED state failed after 1 attempts, retry duration [2]s, error: <nil>. Check MIC pod logs for identity assignment errors\n"
```

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: pkg/auth/auth.go [hunk 1]
```diff
@@ -94,15 +94,15 @@ func (c Config) GetServicePrincipalToken(podName, podNamespace, resource, aadEnd
...
-			return nil, fmt.Errorf("nmi response failed with status code: %d, err: %+v", resp.StatusCode, err)
-		}
-
 		bodyBytes, err := io.ReadAll(resp.Body)
 		if err != nil {
 			return nil, err
 		}
+
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("nmi response failed with status code: %d, err: %+v", resp.StatusCode, string(bodyBytes))
```
Comment (nilekhc) [inline pkg/auth/auth.go:103]: > err: %+v

Should this be `response body` as we are not logging an actual error? Also can we log both err and response body?

Comment (aramase) [reply pkg/auth/auth.go:103]: The error is nil at this point as we're checking for `err != nil` and returning before processing the response.

Comment (aramase) [reply pkg/auth/auth.go]: Updated the string to `response body`

**Review by nilekhc — APPROVED** [review]
lgtm

**Pull request merged**